homeassistant/components/sonos/entity.py

```diff
@@ -70,10 +71,13 @@ async def async_poll(self, now: datetime.datetime) -> None:
                 listener_msg = f"{self.speaker.subscription_address} (advertising as {soco_config.EVENT_ADVERTISE_IP})"
             else:
                 listener_msg = self.speaker.subscription_address
-            _LOGGER.warning(
-                "%s cannot reach %s, falling back to polling, functionality may be limited",
-                self.speaker.zone_name,
-                listener_msg,
+            message = f"{self.speaker.zone_name} cannot reach {listener_msg}, falling back to polling, functionality may be limited"
+            _LOGGER.warning(message)
+            persistent_notification.async_create(
```

Comment:
What should the user do in this case? In my opinion, we should only create notifications if we want the user to perform a specific action. Otherwise it's just a more spammy log message.

Reply:
The problem is that the Sonos devices can not connect back to the Home Assistant host on TCP port 1400 (by default). This is _somewhat_ discussed on the integration page under "[Advanced Use](https://www.home-assistant.io/integrations/sonos/#advanced-use)".

What would you suggest? Improve the documentation to describe the specific problem and link to the page from the notification?

Reply:
Yes, that's good. We should suggest a course of action to take to the user to resolve the problem.

Reply:
Documentation PR here: https://github.com/home-assistant/home-assistant.io/pull/21376

Reply:
Updated log & notification to include links to the proposed docs above.

Reply:
What is a user decides to stick with polling instead?

Reply:
IMO, the polling functionality is a failsafe for when subscriptions fail and should not be used intentionally. If a user has a setup where they won't/can't fix broken subscriptions then they will see this warning on every restart of HA.

Reply:
Ok, that sounds right.

This does make me think about persistent notifications vs logging and makes me wonder if this would be an approach we should take or not. Mainly because it is currently not really common to use a persistent notification for these things?

Reply:
This has had a log warning for several releases and I'm still running into experienced users that are experiencing this problem but haven't noticed the message in the log.

Reply:
My thought is: I need to communicate something to the user in order to ask them to fix something that is affecting the integration and writing warnings/errors in the log isn't enough. This felt like the best method currently available.

Reply:
Brainstorming:
Perhaps the device's communication mode (subscription/polling) can be reported via a sensor (or binary_sensor) and displayed in the device's Sensors card (in the Devices view).

At the very least, it permits the user to create a simple automation to monitor this sensor and report (via whatever notification method they want) when it changes from subscription to polling.

Reply:
A setup should only fall back to polling mode if there is a problem. Once it is fixed by the user it requires no followup actions. There is already a prominent warning in the log but many users are still not aware. This should not need a custom automation to detect a common and known problem state.

Reply:
In that case,  perhaps it should be an integration-level setting where the user can decide if the integration's log messages are echoed as a persistent notification.

Personally, I wouldn't want an integration to, _by default_, report its problems via a persistent notification.

Reply:
> This should not need a custom automation to detect a common and known problem state.

One vote for the binary_sensor.  Those of us who power off our players with smart plugs would like the option to only "monitor" the subset of players we don't power off.  It isn't all or none and it would appear the current approach uses that logic.

Reply:
Perfect. Thanks for the feedback.  It hasn't complained about my powered off players.  Sorry for the noise.

Reply:
> It is not a normal situation to happen

If the trend will be to promote "not normal" situations, from merely being logged to reported as persistent notifications, then all I ask is that the development team consider providing the end-user with basic control over this behavior.

For example, enable it by default but allow the user to disable it (per integration would be best but a global setting is fine). Some users already monitor the log for certain errors and don't need them to also appear in the UI as persistent notifications (where they are visible to all members of the household ... who don't need to see them).